modules/recon/profiles-profiles/namechk.py

```diff
@@ -1,70 +1,40 @@
 from recon.core.module import BaseModule
 from recon.mixins.threads import ThreadingMixin
-from http.cookiejar import CookieJar
-from lxml.html import fromstring
+
 
 class Module(BaseModule, ThreadingMixin):
 
     meta = {
         'name': 'NameChk.com Username Validator',
-        'author': 'Tim Tomes (@LaNMaSteR53) and thrapt (thrapt@gmail.com)',
+        'author': 'Tim Tomes (@LaNMaSteR53) and thrapt (thrapt@gmail.com) and Ryan Hays (@_ryanhays)',
         'version': '1.0',
-        'description': 'Leverages NameChk.com to validate the existance of usernames on specific web sites and updates the \'profiles\' table with the results.',
+        'description': 'Leverages NameChk.com API to validate the existance of usernames on specific web sites and '
+                       'pdates the \'profiles\' table with the results.',
+        'required_keys': ['namechk_api'],
         'comments': (
             'Note: The global timeout option may need to be increased to support slower sites.',
         ),
         'query': 'SELECT DISTINCT username FROM profiles WHERE username IS NOT NULL',
     }
 
     def module_run(self, usernames):
-        # retrieve list of sites
-        self.verbose('Retrieving site data...')
-        url = 'https://namechk.com/'
-        cookiejar = CookieJar()
-        resp = self.request(url, cookiejar=cookiejar)
-        tree = fromstring(resp.text)
-        # extract sites info from the page
-        names = tree.xpath('//div[@class="media record"]/@data-name')
-        labels = tree.xpath('//div[@class="media record"]//h4[@class="media-heading"]/text()')
-        if not len(names) == len(labels):
-            self.error('Inconsistent number of sites and labels.')
-            return
-        # merge names and labels into a list of tuples
-        sites = zip(names, labels)
-        # extract token from the reponse
-        token = ''.join([x.value for x in resp.cookiejar if x.name=='token'])
-        # reset url for site requests
-        url = 'https://namechk.com/availability/%s'
-        payload = {'x': token}
-        # required header for site requests
-        headers = {'X-Requested-With': 'XMLHttpRequest', 'Accept': 'application/json'}
-        for username in usernames:
-            self.heading(username, level=0)
-            payload['q'] = username
-            # validate memberships
-            self.thread(sites, url, payload, headers, cookiejar)
+        key = self.get_key('namechk_api')
+        # Gets a list of available services
+        avail_sites = self.request(url=f"https://api.namechk.com/services/available.json",
```

Comment:
No need to f-string here since you aren't embedding any variables here.

Reply:
Fixed